backend/api/v1/sql_service.go

```diff
@@ -1662,14 +1686,15 @@ func (s *SQLService) prepareRelatedMessage(ctx context.Context, instanceToken st
 		if err != nil {
 			return nil, nil, nil, nil, status.Errorf(codes.Internal, "failed to fetch database: %v", err)
 		}
+
 	}
 
-	environment, err := s.store.GetEnvironmentV2(ctx, &store.FindEnvironmentMessage{ResourceID: &database.EffectiveEnvironmentID})
+	environment, err := s.store.GetEnvironmentV2(ctx, &store.FindEnvironmentMessage{ResourceID: &instance.EnvironmentID})
```

Comment:
`database.EffectiveEnvironment` does not always equal `instance.EnvironmentID`.
We should use `database.EffectiveEnvironment` if `database != nil`, otherwise, use `instance.EnvironmentID`